retail-managment-system/rms-ui/src/app/pages/transaction/transaction.component.ts

```diff
@@ -0,0 +1,147 @@
+import { Component, OnInit } from '@angular/core';
+import {TransactionRepository} from 'src/app/domain/transaction/transaction.repository'
+import { Transaction, TransactionType } from 'src/app/domain/transaction/models/transaction'
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Investors } from '../../domain/investors/models/investor'
+import { InvestorsRepository } from 'src/app/domain/investors/investor.repository';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { MatDialog } from '@angular/material/dialog';
+import { ConfirmDialogComponent } from '../../shared/components/confirm-dialog/confirm-dialog.component';
+
+@Component({
+  selector: 'app-transaction',
+  templateUrl: './transaction.component.html',
+  styleUrls: ['./transaction.component.scss']
+})
+export class TransactionComponent implements OnInit {
+
+  transactionPostForm!: FormGroup
+  allTransactions : Transaction[] = []
+  investorNames : Investors[]=[]
+  transactionTypeEnum = Object.values(TransactionType)
+  displayedColumns: string[] = ['id','investorName','type','amount','date','edit-icon' ,'delete-icon'];
+  currentData!: Transaction
+  isButtonVisible: boolean = true
+  submit: boolean = false;
+  index =0
+
+
+  constructor(
+    private investorsRepository : InvestorsRepository,
+    private transactionRepository : TransactionRepository,
+    private fb :FormBuilder,
+    private SnackBar: MatSnackBar,
+    private dialog: MatDialog
+    ) {
+    
+  }
+
+  ngOnInit(): void {
+    this.getAllTransactions()
+    this.getInvestorData()
+    this.TransactionPostForm()
```

Comment:
name may be more meaningful like TransactionForm
because it isn't post just it is put too